From: Jiri Kosina <jkosina@suse.cz>
Subject: [PATCH] Disable ERST pstore by default
References: bnc#816892
Patch-mainline: not yet

It is reported to cause harmful overwrites of BIOS EEPROM on RX300S7.

Disable it by default until a root-cause (possibly BIOS bug workaround)
is found to avoid machine bricking.

Signed-off-by: Jiri Kosina <jkosina@suse.cz>
---
 fs/pstore/platform.c |    6 ++++++
 1 files changed, 6 insertions(+), 0 deletions(-)

diff --git a/fs/pstore/platform.c b/fs/pstore/platform.c
index 946d061..972953d 100644
--- a/fs/pstore/platform.c
+++ b/fs/pstore/platform.c
@@ -184,6 +184,12 @@ int pstore_register(struct pstore_info *psi)
 		return -EBUSY;
 	}
 
+	if (!backend && !strcmp(psi->name, "erst")) {
+		pr_info("ERST pstore disabled, enforce via pstore.backend=erst");
+		spin_unlock(&pstore_lock);
+		return -EINVAL;
+	}
+
 	if (!backend && !strcmp(psi->name, "efi")) {
 		pr_info("Efi pstore disabled, enforce via pstore.backend=efi");
 		spin_unlock(&pstore_lock);
-- 
1.7.8.3

